### ISSUE TITLE
COMException 0x8007139F in Microsoft.Web.WebView2.WinForms.WebView2.<InitCoreWebView2Async>d__13.MoveNext

### ISSUE THREAD

**Issue by ShaunLoganOracle** [MicrosoftEdge/WebView2Feedback#3008]
**Description**
We have some customers that are encountering this exception when our code is creating a WebView2 control hosted in a Winform. The UserDataFolder used is specific to the user and not shared by other apps. We use the same CoreWebView2Environment properties when calling EnsureCoreWebView2() for subsequent WV2 controls we create. We are unable to reproduce the issue in-house.
```
COMException: The group or resource is not in the correct state to perform the requested operation. (Exception from HRESULT: 0x8007139F)
Source: mscorlib
Stack: 
  at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
  at System.Runtime.InteropServices.Marshal.ThrowExceptionForHR(Int32 errorCode)
  at Microsoft.Web.WebView2.Core.CoreWebView2Environment.<CreateCoreWebView2ControllerAsync>d__41.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
  at Microsoft.Web.WebView2.WinForms.WebView2.<InitCoreWebView2Async>d__13.MoveNext()
```

**Version**
SDK: 1.0.1343.22
Runtime: 107.0.1418.52
Framework: WinForms
OS: Win10, exact version info not yet if available
VSTO Add-in running in Excel

**Repro Steps**
Our code does in essence, the following.

In our Winform that hosts the WV2 control:
```
      private void HandleFormLoaded (object sender, EventArgs e)
      {
            await WebView2Control.InitializeWebView2Async (); // WebView2Control is an instance of our subclass of WebView2
            // once init is complete, we navigate to a URL in the WV2
      }
```
In our subclass of `Microsoft.Web.WebView2.WinForms.WebView2`:
```
      public async Task InitializeWebView2Async ()
      {
         CoreWebView2EnvironmentOptions options = new CoreWebView2EnvironmentOptions ()
         {
            AdditionalBrowserArguments = // <snip> nothing interesting
            Language = // something appropriate,
            AllowSingleSignOnUsingOSPrimaryAccount = true,
         };

         string userDataFolder = // something that resolves to %LocalAppData%\MyCompany\Our Addin Name\
         CoreWebView2Environment webView2Environment = await CoreWebView2Environment.CreateAsync (null, userDataFolder, options);

         await EnsureCoreWebView2Async (webView2Environment); // this is the base class impl: WebView2.EnsureCoreWebView2Async
      }
```

In our event handler for `CoreWebView2InitializationCompleted`, we get `e.IsSuccess` = false with `e.InitializationException` as shown in the stack above.

**Additional context**
We saw this GH issue: #2495, which might be related, but the exception stacks there are WPF so they don't match ours exactly..
We have multiple customers at different companies encountering this issue. We also have many customers across companies **not** encountering this issue.

**Diagnostics**
We are unable to use a debugger at the customer sites.
**How can we capture additional logs from the customer to help us diagnose the root cause of the exception?**

[AB#43230947](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/43230947)

**Comment by ShaunLoganOracle** [comment]
Additional Info: We have one customer that reports getting this exception when they:
1. **Using the Google Chrome browser** Click on a web page link that downloads an XLSX file
2. Open the XLSX file as part of the download operation
3. When Excel opens the file, our add-in is notified and we put up the Winform with the WebView2 control and get the exception: COMException 0x8007139F in Microsoft.Web.WebView2.WinForms.WebView2.<InitCoreWebView2Async>d__13.MoveNext

**Note!** according to the customer, the exception is not seen if they follow these same steps using Firefox, MSIE, and Edge to download + open the XLSX file.

**Comment by ShaunLoganOracle** [comment]
@yunate 
In case it helps, here is a _slightly_ different exception stack from a different customer:
```
COMException: The group or resource is not in the correct state to perform the requested operation. (Exception from HRESULT: 0x8007139F)
Source: mscorlib
Stack: 
  at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
  at Microsoft.Web.WebView2.Core.CoreWebView2Environment.<CreateCoreWebView2ControllerAsync>d__80.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
  at Microsoft.Web.WebView2.WinForms.WebView2.<InitCoreWebView2Async>d__23.MoveNext()
```

**Comment by ShaunLoganOracle** [comment]
@yunate 
re: specific Windows OS versions.  Both of the customers mentioned are on Windows 10.
The first exception stack is from Windows 10 20H2 (19042.2251), using WV2 RT 107.0.1418.62
The second exception stack is from Windows 10 21H2 (19044.2006), also using WV2 RT 107.0.1418.62

I was able to get a the first customer to capture a verbose log from WebView2 by setting:
Env. Var.: `WEBVIEW2_ADDITIONAL_BROWSER_ARGUMENTS = --enable-logging --v=1 --log-file=c:\temp\webview2.log`

I am looking through this log now, is there anything specific I can search for to get closer to what's causing this COMException?

**Comment by ShaunLoganOracle** [comment]
Here is the tail end of the customer's log (before the histograms) showing 3 **ERRORs** (I redacted some information and added blank lines). I know virtually nothing about the internals of Chrome. Are these **ERRORs** significant/related to the COMException? I don't see errors like these when I run WV2 locally.
```
[12556:16216:1202/165413.237:ERROR:device_event_log_impl.cc(215)] [16:54:13.237] USB: usb_device_handle_win.cc:1048 Failed to read descriptor from node connection: A device attached to the system is not functioning. (0x1F)
[12556:16216:1202/165413.241:VERBOSE1:device_event_log_impl.cc(217)] [16:54:13.242] USB: USER: usb_service_win.cc:739 USB device added: path=\\?\usb#vid_04f2&pid_b51c#[REDACTED] vendor=1266 "", product=46364 "", serial="", driver="usbccgp", guid=[REDACTED]

[12556:9728:1202/165413.491:ERROR:util.cc(135)] Can't create base directory: C:\Program Files\Microsoft\EdgeUpdater
[12556:16216:1202/165413.503:VERBOSE1:request_sender.cc(143)] is off the record0
[13896:14660:1202/165413.504:VERBOSE1:network_delegate.cc(34)] NetworkDelegate::NotifyBeforeURLRequest: https://edge.microsoft.com/componentupdater/api/v1/update?cup2key=5:GUtE[REDACTED]
[12556:16216:1202/165413.758:VERBOSE1:request_sender.cc(218)] Request completed from url: https://edge.microsoft.com/componentupdater/api/v1/update?cup2key=5:GUtE[REDACTED]
[12556:16216:1202/165413.760:VERBOSE1:component_updater_service.cc(439)] Update completed with error 0
[12556:16216:1202/165416.141:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Stats.LogStart [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165416.143:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.SystemProfile [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165416.143:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Histograms.Default [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165416.143:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Histograms.Group4 [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165416.143:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Stats.LogComplete [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165419.150:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Stats.LogStart [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165419.150:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.SystemProfile [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165419.150:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Histograms.Default [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165419.151:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Histograms.Group4 [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165419.151:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Stats.LogComplete [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165422.160:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Stats.LogStart [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165422.160:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.SystemProfile [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165422.161:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Histograms.Group4 [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165422.162:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Histograms.Default [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165422.162:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Histograms.Group2 [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165422.162:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Stats.LogComplete [components\telemetry_client\telemetry_service.cc(207)]

[9444:15496:1202/165513.308:ERROR:gpu_init.cc(551)] Passthrough is not supported, GL is disabled, ANGLE is 
[12556:16216:1202/165526.143:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Stats.LogStart [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165526.143:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.SystemProfile [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165526.144:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Histograms.Group4 [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165526.144:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Histograms.Default [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165526.144:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Histograms.Group2 [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165526.144:VERBOSE1:telemetry_service.cc(626)] NotInSampleGroup: Microsoft.WebBrowser.Protobuf.UMA.Stats.LogComplete [components\telemetry_client\telemetry_service.cc(207)]
[12556:16216:1202/165526.148:VERBOSE1:sync_session_durations_metrics_recorder.cc(95)] Session end
[12556:16216:1202/165526.149:VERBOSE1:account_reconcilor.cc(225)] AccountReconcilor::Shutdown
[12556:16216:1202/165526.149:VERBOSE1:account_reconcilor.cc(659)] AccountReconcilor::AbortReconcile: try again later
[12556:16216:1202/165526.149:VERBOSE1:account_reconcilor.cc(756)] AccountReconcilor::CalculateIfMultiloginReconcileIsDone: Error was 
[12556:16216:1202/165526.149:VERBOSE1:account_reconcilor.cc(268)] AccountReconcilor::UnregisterWithIdentityManager
[12556:16216:1202/165526.149:VERBOSE1:account_reconcilor.cc(247)] AccountReconcilor::UnregisterWithContentSettings
[12556:16216:1202/165526.149:VERBOSE1:gaia_cookie_manager_service.cc(707)] GaiaCookieManagerService::CancelAll
[12556:16216:1202/165526.149:VERBOSE1:mutable_profile_oauth2_token_service_delegate.cc(822)] MutablePO2TS::Shutdown
[12556:16216:1202/165526.149:VERBOSE1:mutable_profile_oauth2_token_service_delegate.cc(321)] MutablePO2TS::~MutablePO2TS
[12556:16216:1202/165526.149:VERBOSE1:device_event_log_impl.cc(217)] [16:55:26.149] Bluetooth: EVENT: bluetooth_api.cc:109 BluetoothAPI: Shutdown
[12556:16216:1202/165526.153:VERBOSE1:account_reconcilor.cc(168)] AccountReconcilor::~AccountReconcilor
[12556:16216:1202/165526.153:VERBOSE1:device_event_log_impl.cc(217)] [16:55:26.153] Bluetooth: EVENT: bluetooth_api.cc:95 ~BluetoothAPI: 00003B7000298600
[13896:14940:1202/165526.171:VERBOSE1:statistics_recorder.cc(520)] Collections of all histograms
```

**Comment by sln162** [comment]
My two users encountered the same error 0x8007139F during initialization. The runtime versions of WebView 2 are 106.0.1370.47 and 107.0.1418.35.
In this case, if:
1. Reinitialize Now
2. Wait a few seconds for initialization
3. Initialize after deleting a folder (create a new folder if an exception occurs during deletion)

Since it is difficult for my users to get in touch with them, I am curious whether the above has any effect

**Comment by sln162** [comment]
to update:
At present, I use a new folder directory, re initialize it, and work normally. According to preliminary analysis, it may be related to the software restart, because the mode of starting itself first and then exiting (clean exit) is adopted, which may lead to the problem that the folder is released by another process when it is occupied. Of course, it's just a guess. It's always good to add retry when initialization fails

**Comment by AlexDavisOracle** [comment]
@yunate are you looking at this issue actively? It is a hot topic for us.
Please review the posts from Shaun above and provide some update. Thanks!

**Comment by AlexDavisOracle** [comment]
@nishitha-burman Hi! This topic is pretty hot for us currently. Would it be possible to get someone to look at it? Thanks!

**Comment by ShaunLoganOracle** [comment]
I received a WebView2 log from another customer (also via by Env. Var.: `WEBVIEW2_ADDITIONAL_BROWSER_ARGUMENTS = --enable-logging --v=1 --log-file=c:\temp\webview2.log`). In this case the only error in the log is:
```
[9016:19016:1212/144338.629:ERROR:edge_auth_errors.cc(477)] EDGE_IDENTITY: Get Default OS Account failed: Error: Primary Error: kImplicitSignInFailure, Secondary Error: kAccountProviderFetchError, Platform error: 0, Error string: 
```
However I see the same error when I run the add-in w/WebView2 locally - and the COMException 0x8007139F does not repro for me.  Since we are not able to run a debugger at the customer sites where this issue occurs, **is there some additional logging or diagnostic information we can capture** to help us find what is causing this exception?

**Comment by AlexDavisOracle** [comment]
@champnic would you be able to help with this issue? Thanks!

**Comment by ShaunLoganOracle** [comment]
For anyone else following this issue, I stumbled upon this reference to getting WV2-specific event logged in the Event Viewer: [link](https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/diagnostics/etw.md)

**Comment by yunate** [comment]
@ShaunLoganOracle, I'm sorry I haven't had one for so long, and I will look into this issue.

**Comment by yunate** [comment]
@ShaunLoganOracle  Is it possible to take a perf trace for this? The instruction on how to take a trace is at https://github.com/microsoft/microsoft-ui-xaml/issues/5816#issuecomment-916357593. The trace file will be a large file that has to be shared via some cloud drive.

**Comment by ShaunLoganOracle** [comment]
@yunate 
Thanks for the reply.  I notice that two different WPRP files have been suggested:
1. EdgeWebView2_CPU.wprp from [here](https://github.com/microsoft/microsoft-ui-xaml/issues/5816#issuecomment-916357593)
2. WebView2.wprp from [here](https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/diagnostics/etw.md)
Which one is best for this case?

**Comment by ShaunLoganOracle** [comment]
Capturing WV2-specific events seems like it would be helpful. Unfortunately, we are blocked from using the Windows Performance Recorder because:
1. We (and our customers) are using Windows 10.  After installing the appropriate WPR from [here](https://learn.microsoft.com/en-us/windows-hardware/get-started/adk-install#other-adk-downloads), we are unable to successful capture a recording file.  We get (on the `wpr -stop` command):
```
        Cannot change thread mode after it is set.
        Profile Id: RunningProfile
        Error code: 0x80010106
```
2. This [MS blog post](https://devblogs.microsoft.com/performance-diagnostics/wpr-start-and-stop-commands/) indicates that the way around this error is to install a later build of the WPR ("build 19650 or later"), from the Insider Channel.  Our customers may not be Windows Insiders, so this is not a viable approach.

For others reading this, Admin/elevated privileges are required to install/run the WPR. This may not be feasible in all cases.

**Is there another mechanism for getting a newer Windows ADK for Windows 10 (with the WPR) that will successfully record a trace file?**

**Comment by yunate** [comment]
> Capturing WV2-specific events seems like it would be helpful. Unfortunately, we are blocked from using the Windows Performance Recorder because:
> 
> 1. We (and our customers) are using Windows 10.  After installing the appropriate WPR from [here](https://learn.microsoft.com/en-us/windows-hardware/get-started/adk-install#other-adk-downloads), we are unable to successful capture a recording file.  We get (on the `wpr -stop` command):
> 
> ```
>         Cannot change thread mode after it is set.
>         Profile Id: RunningProfile
>         Error code: 0x80010106
> ```
> 
> 2. This [MS blog post](https://devblogs.microsoft.com/performance-diagnostics/wpr-start-and-stop-commands/) indicates that the way around this error is to install a later build of the WPR ("build 19650 or later"), from the Insider Channel.  Our customers may not be Windows Insiders, so this is not a viable approach.
> 
> For others reading this, Admin/elevated privileges are required to install/run the WPR. This may not be feasible in all cases.
> 
> **Is there another mechanism for getting a newer Windows ADK for Windows 10 (with the WPR) that will successfully record a trace file?**

For this issue, we could use DebugView(https://learn.microsoft.com/en-us/sysinternals/downloads/debugview) to find the log like WebView2: Initialization failed.
image

**Comment by ShaunLoganOracle** [comment]
@yunate Thank you. I want to give the customer who can reproduce the issue the correct instructions, can you please confirm whether or not the default Capture settings for DebugView are sufficient to capture the WebView2 output?
![image](https://user-images.githubusercontent.com/17167099/209977818-e667fb4f-52ac-4bb4-b5f4-4334de5f4426.png)

**Comment by yunate** [comment]
> 

Fortunately, we have logged by OutputDebugString function specially for except 0x8007139F, so we can use DebugView to capture the WebView2 output and use default Capture settings is okay.

**Comment by ShaunLoganOracle** [comment]
@yunate We received a DebugView log from one of the customers.  It contains:
```
00002219	46.17500305	[3040] RecursiveDirectoryCreate( C:\Users\<USER>\AppData\Local\Oracle\ADF Desktop Integration\EBWebView directory exists )	
00002220	46.68103027	[18140] WebView2: Initialization failed due to a mismatch in DPI awareness. Please ensure that the DPI awareness of the host app matches the DPI awareness of current WebView2 processes using the same user data folder.
```
In our case, we are a VSTO add-in running inside of the Excel.exe process. We don't control the DPI awareness of the host app, and we're getting this exception with a single WebView2 process associated with the UDF (unlike #899 where there are two apps w/different DPI awareness sharing the UDF).

Since we re-use the UDF for different excel.exe sessions, could this exception be caused by the user running Excel (and WV2) on set of display settings, then running Excel again with different settings (but sharing the same UDF)?
Are there other causes that would explain this debug log message?

**Comment by ShaunLoganOracle** [comment]
@yunate We got another DebugView log from another customer, it shows the same "... mismatch in DPI awareness..." issue.
What could be the cause of this error in our case?

**Comment by yunate** [comment]
> @yunate We received a DebugView log from one of the customers. It contains:
> 
> ```
> 00002219	46.17500305	[3040] RecursiveDirectoryCreate( C:\Users\<USER>\AppData\Local\Oracle\ADF Desktop Integration\EBWebView directory exists )	
> 00002220	46.68103027	[18140] WebView2: Initialization failed due to a mismatch in DPI awareness. Please ensure that the DPI awareness of the host app matches the DPI awareness of current WebView2 processes using the same user data folder.
> ```
> 
> In our case, we are a VSTO add-in running inside of the Excel.exe process. We don't control the DPI awareness of the host app, and we're getting this exception with a single WebView2 process associated with the UDF (unlike #899 where there are two apps w/different DPI awareness sharing the UDF).
> 
> Since we re-use the UDF for different excel.exe sessions, could this exception be caused by the user running Excel (and WV2) on one set of display settings, then running Excel again with different settings (but sharing the same UDF)? Are there other causes that would explain this debug log message?

Thanks for your reply, and I'm the reason of DPI issue, and I will inform you of any progress

**Comment by yunate** [comment]
This is a bug and will be fix in next release (release-109), and this will be published soon.

**Comment by AlexDavisOracle** [comment]
@yunate Great news! Thanks for the update! Will it be fixed in the WebView2 **runtime** or **SDK**?

**Comment by yunate** [comment]
> @yunate Great news! Thanks for the update! Will it be fixed in the WebView2 **runtime** or **SDK**?

Yes, both runtime and SDK will be all updates. And should be published next week.

**Comment by yunate** [comment]
It requires an SDK upgrade to make the app can use the 109 runtime.

**Comment by yunate** [comment]
The SDK version should match the runtime. The SDK file: `Microsoft.Web.WebView2.Wpf.dll/Microsoft.Web.WebView2.Core.dll` should be upgraded. If you do not upgrade the SDK, the runtime you used may be the old one.

**Comment by ShaunLoganOracle** [comment]
@yunate 
It is great news that the fix has been identified! In the meantime, is there any action we can have our customers take to avoid/workaround this issue (until they get the fix)?

**Comment by ShaunLoganOracle** [comment]
@yunate 
One of customers that is getting the COMException 0x8007139F upgraded to WV2 Runtime 109.0.1518.52, deleted their UDF, and was still able to reproduce the issue. We have diagnostic logging that confirms the version of the WV2 RT in use when they reproduced the issue.
Is this fix in the WV2 Runtime or the WV2 SDK?

**Comment by yunate** [comment]
> @yunate One of customers that is getting the COMException 0x8007139F upgraded to WV2 Runtime 109.0.1518.52, deleted their UDF, and was still able to reproduce the issue. We have diagnostic logging that confirms the version of the WV2 RT in use when they reproduced the issue. Is this fix in the WV2 Runtime or the WV2 SDK?

Thank for you feedback!
This fix in the WV2 Runtime, the app should not need to do some update but install the `1.0.1518.46` Edge (`1.0.1518.46` runtime). could you help to confirm if the customer's version is 1.0.1518.46, the fix should have published in `1.0.1518.46`. If the customer's version is 1.0.1518.46, and can still able to reproduce the issue. I will raise this issue again in our teams.

**Comment by AlexDavisOracle** [comment]
> This fix in the WV2 Runtime, the app should not need to do some update but install the `1.0.1518.46` Edge (`1.0.1518.46` runtime). could you help to confirm if the customer's version is 1.0.1518.46, the fix should have published in `1.0.1518.46`.

I am under the impression that WebView2 is independent of Edge (the browser). Are you saying that here is some dependency between them? @yunate

**Comment by yunate** [comment]
> 

They are independent of Edge, can install only the runtime. The version of runtime will be released with Edge, so I said to install Edge. Of couse only install the runtime has no problem.

**Comment by AlexDavisOracle** [comment]
Confused...

We are focused on WebView2. You said the issue was fixed in the "109 runtime". The customer upgraded the WebView2 Runtime to 109.0.1518.52. The problem still occurs.

Is the fix in 109.0.1518.52? Or, do we need to wait for a later version?

@yunate

**Comment by AlexDavisOracle** [comment]
In case it matters, the customer upgraded WebView2 from here:
https://developer.microsoft.com/en-us/microsoft-edge/webview2/consumer/

**Comment by yunate** [comment]
> Confused...
> 
> We are focused on WebView2. You said the issue was fixed in the "109 runtime". The customer upgraded the WebView2 Runtime to 109.0.1518.52. The problem still occurs.
> 
> Is the fix in 109.0.1518.52? Or, do we need to wait for a later version?
> 
> @yunate
Sorry for confused.
Other users also found this problem, They solved the problem by updating the version. https://github.com/MicrosoftEdge/WebView2Feedback/issues/3068
But it seems this do not fix your problem. This may be that the causes of these two problems are different. We need to continue to investigate.
And by the way, I want to ensure the customer has used the latest version. It's possible that the latest version has installed but not actually used.
![image](https://user-images.githubusercontent.com/37793342/213068158-18834fb2-0f6f-42ac-9d71-7025830f30fc.png)

**Comment by ShaunLoganOracle** [comment]
@yunate 
In our add-in, we call Microsoft.Web.WebView2.Core.CoreWebView2Environment.GetAvailableBrowserVersionString() and log the result.  For our customer that is still experiencing the issue, we see "109.0.1518.52".

**Comment by LiangTheDev** [comment]
Currently, the only known cause of dpi awareness causing webview init failure is app compat setting forcing msedgewebview2.exe to run with a different dpi awareness. Let's wait for customers to get back the contents in `HKCU\SOFTWARE\Microsoft\Windows NT\CurrentVersion\AppCompatFlags`.

**Comment by ShaunLoganOracle** [comment]
@LiangTheDev Thank you.
Another data point in case it is significant:
On my Win 10 dev box (issue does not occur), I exposed the DPI Awareness column in Task Manager. The EXCEL.EXE process shows `System`.  The 7 msedgewebview2.exe processes show 2 different values in that column: the main process (with `--webview-exe-name=EXCEL.EXE`) shows `System` as does the process with `--type=gpu-process`. The others show `Unaware` for DPI Awareness (including the one with `--type=renderer`).  Is this variety for DPI Awareness among the WV2 processes expected?

**Comment by LiangTheDev** [comment]
That's expected. DPI awareness only matters when the processes need to know dpi. Processes like renderer process is not expected to interact with the system.

**Comment by ShaunLoganOracle** [comment]
@LiangTheDev 
We heard back from our customer. Unfortunately their IT policies prevent them from exporting the HKCU key (HKCU\SOFTWARE\Microsoft\Windows NT\CurrentVersion\AppCompatFlags).  They had an admin user export it, but of course, those values don't reflect the user having the issue.  I have attached that file anyway in case there's a clue there (there is no layers\ subkey AFAICT).
I also asked the user (who is seeing the DPI mismatch issue) to capture the DPI settings from Excel.exe and msedgewebview2.exe.  Those 2 screen shots are also attached.
[app-flags-redacted.txt](https://github.com/MicrosoftEdge/WebView2Feedback/files/10469504/app-flags-redacted.txt)
![DPI_Settings_EXCEL EXE](https://user-images.githubusercontent.com/17167099/213789140-099870e6-96cf-41ff-89bd-9bdc4c4eb790.jpg)
![DPI_Settings_msedgewebview2 exe](https://user-images.githubusercontent.com/17167099/213789143-c910e193-fb65-48a7-a5e3-8fe1d5a2c9c5.jpg)

I am not familiar with the DPI Awareness topic, are there other ways of confirming how the incompatibility arises (other configuration to examine)?  For example would HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\AppCompatFlags perhaps show something interesting?

**Comment by LiangTheDev** [comment]
Things under HKLM would has the same influence as HKCU. But the screen shot suggests that it is at least not the same normal app compat issue we have been seeing.
I am not familiar with all the dpi awareness details either. So, not much idea here.
Not sure whether the user is allowed or able to do these. The user could try to check if the any other WebView2 application can run. Then we could try to check dpi awareness of those msedgewebview2.exe and whether apphelp.dll is loaded in those processes. If the user can run the [sample app](https://github.com/MicrosoftEdge/WebView2Samples/tree/main/SampleApps/WebView2APISample), he/she could run it with different dpi awareness and see whether the issue could also repro with the sample app. For the sample app, use command line switch --dpisystemaware to run as system aware and --dpiunaware to run as unaware.

**Comment by ShaunLoganOracle** [comment]
@LiangTheDev Thank you.  In the meantime we received the AppCompatFlags export from a different customer experiencing the issue.  In their case, they have:
```[HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows NT\CurrentVersion\AppCompatFlags\Layers]
"C:\\Program Files\\Microsoft Office\\root\\Office16\\EXCEL.EXE"="~ GDIDPISCALING DPIUNAWARE"
```
And the screen shot of Excel.exe's compatibility/DPI Awareness shows:
![image](https://user-images.githubusercontent.com/17167099/214134951-8c03862c-baf4-4ebe-9f5a-3efb1ec28b83.png)
Just to confirm, does this suggest that this customer is experiencing what you called
>the same normal app compat issue we have been seeing.

?

**Comment by LiangTheDev** [comment]
Not necessarily the same issue as we experienced previously.
The known issue is system forcing msedgewebview2.exe to be a certain DPI awareness and therefore causing the WebView2 code failing to set desired dpi awareness for msedgewebview2.exe processes.
In this case, the app compat is forcing excel.exe to be a certain DPI awareness. WebView2 code should be able to adapt to the real dpi awareness of the hosting process. So, if the user data folder is not shared with any other app, we still don't know why it fails.
Just to make sure, the issue is not with an Excel add-in where WebView2 hosting is provided by Office, right? I ask because Excel 's built in WebView hosting code "knows" about its dpi awareness and that code might not work well when OS forces the process to be in a different dpi awareness.
To make sure we understand the basic scenario correctly, if possible, could you ask the user to take an ETW trace so that we know whether there is multiple WebViews involved and what dpi awareness WebView2 code is trying to use? The instruction for taking ETW trace: https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/diagnostics/etw.md.

**Comment by ShaunLoganOracle** [comment]
>the issue is not with an Excel add-in where WebView2 hosting is provided by Office, right?

If I am understanding you correctly: No, our add-in hosts the WV2 control using Windows.Forms.
In other words, our code is a VSTO add-in loaded into the Excel.exe process.  Our add-in creates a `System.Windows.Forms.Form` modal dialog instance that hosts/contains a `Microsoft.Web.WebView2.WinForms.WebView2` control.

The User Data Folder is not shared with any other apps.  It _is_ shared between subsequent launches of Excel.exe (and so WV2) but we always use the same environment settings when initializing the WV2 control (and we've confirmed that the issue repros the first time - when no UDF exists).

By altering the Properties on EXCEL.EXE to override the High DPI scaling to be **System (Enhanced)**, we can now reproduce the same COMException locally that one of our customers is seeing with that same override (seen in the most recent screen shot above).

FYI: We also have another VSTO add-in that uses WPF (not Winforms) for hosting the WebView2 control where we can also repro the same COMException if we override the High DPI scaling behavior in the same way.

I previously tried getting an ETW trace but was getting errors from WPR, like:
```
        Cannot change thread mode after it is set.
        Profile Id: RunningProfile
        Error code: 0x80010106
```
I'll give that another try.

I will also use Process Explorer to confirm whether apphelp.dll is loaded by Excel and/or msedgewebview2 when reproducing the issue.

Is this a fair summary?
The issue that you know about:
https://github.com/MicrosoftEdge/WebView2Feedback/issues/3029#issuecomment-1349131747
which appears to be fixed in the 109 WV2 RT appears to have a different root cause than the issue I can reproduce.
"Your" issue is:

>If you see "Initialization failed due to a mismatch in DPI awareness" error message, and apphelp.dll loaded into msedgewebview2.exe, it is almost certain that the failure was due to app compat forcing msedgewebview2.exe to run with different dpi awareness than the app

Our issue can be triggered by overriding high DPI awareness for Excel.exe - we don't see anything in the registry forcing msedgewebview2.exe to run with any specific dpi awareness..

**Comment by LiangTheDev** [comment]
Your summary is correct. This is a different issue from what we know about. I have not been able to repro with WPF WebView2 sample app, so an ETW trace for the repro would be appreciated.

**Comment by ShaunLoganOracle** [comment]
@LiangTheDev 
I think we are dealing with 3 issues here: 
**Issue1:** the issue you know about 
**Issue2:** our "customer1" issue where EXCEL.EXE had high dpi awareness overridden, I am able to repro Issue2 locally.
**Issue3:** our "customer2" issue where there are no overrides for high dpi awareness on EXCEL.EXE or msedgewebview2.exe. I am not able to repro Issue3 locally.

Our "customer1" has resolved the issue by removing the override from EXCEL.EXE.

Our "customer2" is still experiencing the COMException.  So we're focusing on trying to get to the root cause of that issue.  We are hampered by the customer not having admin rights on their computer. Their registry is pretty locked down too so they cannot get the export for HCKU\...\AppCompatFlags.  They _were_ able to inspect properties on EXCEL.EXE and msedgewebview2.exe to confirm no dpi overrides. Without admin rights, they won't be able to get the ETW trace when reproducing the issue. Per [the instructions](https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/diagnostics/etw.md), wpr must be run in an elevated command prompt.
Both customers are using our Excel add-in and Winforms to host the WV2 control.
Any other thoughts about getting diagnostics from the customer for Issue3?

**Comment by LiangTheDev** [comment]
I don't have any ideas on how to move investigation of issue3 further. I'll ask around to see whether other people have any ideas.

Since you can repro issue2, and issue2 and issue3 might have the same cause, could we try to get an ETW trace for issue2, or get a simple repro project/steps so that we can also repro it?

**Comment by LiangTheDev** [comment]
@ShaunLoganOracle , it seems that we exporting registry key doesn't require running as admin. Have you tried to ask the user to run something like the following from the cmd.exe (the result will be in appCompatReg.txt)?
`reg export "HKCU\SOFTWARE\Microsoft\Windows NT\CurrentVersion\AppCompatFlags" appCompatReg.txt`

**Comment by ShaunLoganOracle** [comment]
>  Have you tried to ask the user to run something like the following from the cmd.exe (the result will be in appCompatReg.txt)?
reg export "HKCU\SOFTWARE\Microsoft\Windows NT\CurrentVersion\AppCompatFlags" appCompatReg.txt

@LiangTheDev 
Yes, we had incorrect info from the customer, they are getting that export now. Thanks.
I am still working on getting an ETW trace for the one case I can repro.

**Comment by ShaunLoganOracle** [comment]
I am preparing a sample add-in that repros Issue2.

**Comment by ShaunLoganOracle** [comment]
@LiangTheDev 
I have the sample add-in ready, may I send it to you via email?

**Comment by LiangTheDev** [comment]
@ShaunLoganOracle , please share the sample add-in along with repro instructions to lzhao@microsoft.com.

**Comment by LiangTheDev** [comment]
Thanks @ShaunLoganOracle for sharing the repro project. I can repro it and issue2 has the same root cause as issue1.
It seems that app compat is “smart” enough to also apply the settings to child msedgewebview2.exe processes in Excel case. I didn’t know this. The rest is the same as the issue which I know previously, app compat forcing msedgewebview2.exe to be of a different dpi awareness and WebView2 code detect it as a mismatch and refuses to continue webview creation.

We’ll investigation what we can do about it. For customers, the workaround would be to remove the app compat setting for the app as well (Excel in this case).

To collect diagnostic info, you could ask customers to collect the app compat settings, and collect the debug logs by setting environment variable WEBVIEW2_ADDITIONAL_BROWSER_ARGUMENTS with value of --enable-logging --v=1 --log-file=log-file-path, like

1. run from cmd.exe: setx WEBVIEW2_ADDITIONAL_BROWSER_ARGUMENTS "--enable-logging --v=1 --log-file=c:\temp\webview2-test.log"
2. Repro
3. Run from cmd.exe to clear the environment variable: setx WEBVIEW2_ADDITIONAL_BROWSER_ARGUMENTS ""
4. Share the webview2-test.log

**Comment by ShaunLoganOracle** [comment]
> I can repro it and issue2 has the same root cause as issue1 ... We’ll investigat[e] what we can do about it.

Good to know.  Please keep us updated on the progress.

> the workaround would be to remove the app compat setting

Yes, thanks. Our customer experiencing Issue2 has taken that approach.

That leaves us with Issue3 (same COMException, but no override seen for Excel.exe's DPI setting.  This customer was able to get the HKCU app compat settings; I will send them via email. I did not see anything pertinent, although there is an old setting under Layers: "C:\\Program Files (x86)\\Microsoft Office\\**Office15**\\EXCEL.EXE"="~ HIGHDPIAWARE"
I don't think that applies since they are running EXCEL.EXE from **\\Office16\\**.

>collect the debug logs by setting environment variable WEBVIEW2_ADDITIONAL_BROWSER_ARGUMENTS

This customer previously collected that log, but I did not see anything that jumped out, I'll send it via email too.

For this customer (Issue3), the log captured from DebugView showed the same error as in Issue2:
`WebView2: Initialization failed due to a mismatch in DPI awareness. Please ensure that the DPI awareness of the host app matches the DPI awareness of current WebView2 processes using the same user data folder.`

I'll send the whole log as well.

**Comment by ShaunLoganOracle** [comment]
Hi @LiangTheDev ,
We heard back from the customer experiencing Issue 3.  As mentioned, EXCEL.EXE did not have a high DPI override.  **However**, their use case differed slightly from the customer seeing Issue 2.  For Issue 3, the customer:

1. Opens a Chrome browser
2. Downloads a workbook file (.XLSX)
3. In the lower left corner of Chrome's window, clicks on the downloaded XLSX file to open it (in Excel)
4. ... which then triggers our add-in to open the Winform hosting the WV2 control (which fails during init with the COMException 0x8007139F).

We found that their CHROME.EXE file properties > Compatibility > ... had the Override high DPI scaling behavior checkbox checked (Scaling performed by: Application). Unchecking it avoids the COMException.  I don't know if this makes technical sense, but logically, the DPI override seems to be getting passed along from Chrome.exe to Excel.exe and then causes the issue in msedgewebview.exe.

We look forward to hearing back from you about a resolution for Issue 2 with the hope that it will also resolve Issue 3.

**Comment by LiangTheDev** [comment]
Thanks for helping us understanding the issue. We'll see what we can do about it.

**Comment by ShaunLoganOracle** [comment]
In case it helps, I was able to repro Issue 3 locally using MS Edge (where msedge.exe had high DPI scaling override set) and the sample add-in I previously provided.  Download an XLSX file using MS Edge, click the link in the upper left, Excel opens, then trigger the add-in to open the Winform w/WV2 (Add-ins > Run Test).  In this case, Excel.exe **did not** have any DPI override.

**Comment by ShaunLoganOracle** [comment]
Hi @LiangTheDev 
It has been a while and we still have customers encountering this issue (and asking us for updates about a fix). Is there anything I can pass along to them?
Thanks.

**Comment by G-Equipment** [comment]
Hello,

I have the same probleme here, nothing special in the regedit.
Here is the base error : 
`
UNHANDLED EXCEPTION IN MAIN APPLICATION THREAD: [COMException]: "Opération abandonnée (Exception from HRESULT: 0x80004004 (E_ABORT))"
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   at Microsoft.Web.WebView2.Core.CoreWebView2Environment.<CreateCoreWebView2ControllerAsync>d__88.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Web.WebView2.WinForms.WebView2.<InitCoreWebView2Async>d__26.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at IHM_v8.WebBrowser.<PdfV_Load>d__16.MoveNext() in C:\dev\IHM_v8\IHM_v8\WebBrowser.cs:line 118
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() 
`

Here is the log from WEBVIEW2_ADDITIONAL_BROWSER_ARGUMENTS : 

`
[14088:12960:0428/171541.219:WARNING:chrome_main_delegate.cc(653)] This is Chrome version 112.0.1722.58 (not a warning)
[14088:12960:0428/171541.285:WARNING:chrome_browser_cloud_management_controller.cc(87)] Could not create policy manager as CBCM is not enabled.
[14088:12960:0428/171541.288:VERBOSE1:variations_field_trial_creator.cc(88)] VariationsSeedUsage:2
[14088:12960:0428/171541.288:VERBOSE1:variations_field_trial_creator.cc(443)] VariationsSetupComplete
[14088:12960:0428/171541.300:ERROR:win_util.cc(1496)] failed to set WebView dpi awareness of 1: L’opération a réussi. (0x0)
[14088:12960:0428/171541.329:VERBOSE1:webrtc_event_log_manager.cc(97)] WebRTC remote-bound event logging enabled.
[14088:12960:0428/171541.330:VERBOSE1:pref_proxy_config_tracker_impl.cc(187)] 00003EE0022EC900: set chrome proxy config service to 00003EE0022C8540
[14088:8888:0428/171541.351:VERBOSE1:media_stream_manager.cc(1596)] MSM::InitializeMaybeAsync([this=00003EE002286100])
[14088:8888:0428/171541.351:VERBOSE1:media_stream_manager.cc(1596)] MDM::MediaDevicesManager()
[14088:8888:0428/171541.351:VERBOSE1:media_stream_manager.cc(1596)] MSM::MediaStreamManager([this=00003EE002286100]))
[14088:20612:0428/171541.353:VERBOSE1:chrome_browser_main.cc(779)] TryInitWindowTabManager start
[14088:20612:0428/171541.353:VERBOSE1:wtm_interface_factory.cc(85)] WTM is not used for WebView.
[14088:12960:0428/171541.355:ERROR:chrome_browser_cloud_management_controller.cc(162)] Cloud management controller initialization aborted as CBCM is not enabled.
`

**Comment by ShaunLoganOracle** [comment]
@G-Equipment Are you sure that you're seeing the same issue? I reported COMException `0x8007139F`. You're seeing  `0x80004004 (E_ABORT)`.  If you search for E_ABORT in this form ( [is:issue E_ABORT](https://github.com/MicrosoftEdge/WebView2Feedback/issues?q=is%3Aissue+E_ABORT) ) you'll see other issues with that error code from Winforms.

**Comment by xiakan** [comment]
@ShaunLoganOracle I have also encountered the same problem as you. Have you resolved this issue?

**Comment by ShaunLoganOracle** [comment]
> @ShaunLoganOracle I have also encountered the same problem as you. Have you resolved this issue?

@xiakan 
No, please see @LiangTheDev 's reply from April 10 in this thread

**Comment by xiakan** [comment]
Restarting the computer suddenly throws an exception, and the usual recovery method is to restart the computer again

**Comment by LiangTheDev** [comment]
@xiakan it is the same issue and that issue is not resolved if you see this debug message when attaching the debugger to the application process: "WebView2: Initialization failed due to a mismatch in DPI awareness.".
It is related to app compat layer applied to msedgewebview2.exe, the application exe or some ancestor process of the application process. For this issue, we currently don't have a solution. Not sure why restarting the computer would change app compat settings. So, please first make sure that it is the same issue.

**Comment by ShaunLoganOracle** [comment]
> April 10:
> @ShaunLoganOracle unfortunately there is no meaningful progress made on this. I've made the issue more visible and hopefully we could make some progress in the near future.

@LiangTheDev some of our joint (Microsoft & Oracle) customers are inquiring about progress on this issue. Is there anything I can share with them?

**Comment by LevchenkoOleksandr** [comment]
I also have the same issue on my dev machine: (The group or resource is not in the correct state to perform the requested operation)
I have Outlook COM add-in hosing WebView2 (Win32/WPF)
Latest runtime **114.0.1823.82**
Have similar issue on **DebugView**:
WebView2: Initialization failed due to a mismatch in DPI awareness. Please ensure that the DPI awareness of the host app matches the DPI awareness of current WebView2 processes using the same user data folder
No compatibility overrides for Outlook.

Ready to cooperate for solving the issue since I have dev machine with all possible access and permissions

**Comment by LiangTheDev** [comment]
We just made a change for the dpi awareness issue. Could you please try it out with Edge Canary and see whether the fix works? The change is in Edge version >= 117.0.2007.0, which means available in latest Edge Canary only currently.

**Comment by LevchenkoOleksandr** [comment]
Hello support team. Unfortunately it didn't fix in my case. 
I have downloaded Canary Edge version 117.0.2007.0 and tried to initialize my webview2 with direct path:
**result** = await CoreWebView2Environment.CreateAsync("C:\Users\[my_user_name]\AppData\Local\Microsoft\Edge SxS\Application\117.0.2007.0", ...

Ensured loaded version  **result**.BrowserVersionString is 117.0.2007.0 canary
And got same warning:  **WebView2: Initialization failed due to a mismatch in DPI awareness. Please ensure that the DPI awareness of the host app matches the DPI awareness of current WebView2 processes using the same user data folder.**

As the result I have initial error:
**The group or resource is not in the correct state to perform the requested operation. (Exception from HRESULT: 0x8007139F**

**Comment by LiangTheDev** [comment]
@LevchenkoOleksandr is there anything under Software\Microsoft\Windows NT\CurrentVersion\AppCompatFlags\Layers?

**Comment by LevchenkoOleksandr** [comment]
No, I didn't set any compatibility overrides

**Comment by LiangTheDev** [comment]
@LevchenkoOleksandr then it is not the same issue discussed in this thread and the fix is trying to fix. Could you please take an ETW trace of the repro? Instruction for taking ETW trace is at https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/diagnostics/etw.md, using https://github.com/MicrosoftEdge/WebView2Feedback/blob/main/diagnostics/WebView2.wprp should be good enough. You can share the trace privately with me at lzhao@microsoft.com.

**Comment by ShaunLoganOracle** [comment]
@LiangTheDev , you said:
> We just made a change for the dpi awareness issue. Could you please try it out with Edge Canary and see whether the fix works? The change is in Edge version >= 117.0.2007.0, which means available in latest Edge Canary only currently.

I reproduced the issue using the simple use case of changing High DPI scaling on Excel.exe to "System (Enhanced)", using WV2 RT `115.0.1901.188`.  Then I uninstalled that WV2 RT and installed the Edge Canary (`117.0.2017.0 canary`).
Good news: when I tried my simple case, the issue did not reproduce.  So, I am hopeful that once the 117.0.xxx WebView2 Runtime gets pushed out to customers, they will no longer encounter this issue.

**Comment by ShaunLoganOracle** [comment]
Since we have customers waiting for this fix, is there anything you can say about the timeline for it appearing (even roughly)?

**Comment by LiangTheDev** [comment]
@ShaunLoganOracle thanks for testing it and great to hear that the change seems to make a difference. Edge 117 should be released around Week of 14-Sep-2023. 

And in offline discussion, it is found that the issue @LevchenkoOleksandr encountered is likely due to multiple apps sharing the same user data folder.

**Comment by LevchenkoOleksandr** [comment]
> And in offline discussion, it is found that the issue @LevchenkoOleksandr encountered is likely due to multiple apps sharing the same user data folder.

Thank you for your help and pointing me to correct direction. In my case it really was sharing the same folder between components.

**Comment by ShaunLoganOracle** [comment]
>Edge 117 should be released around Week of 14-Sep-2023

I look forward to trying out the fix soon.
For the record, the issue still repros with SDK 1.0.1938.49 and RT 116.0.1938.76

**Comment by LiangTheDev** [comment]
As with all software development, release schedule is often delayed. From information I have gathered, it looks like Edge 117 is likely to be released sometime next week.

**Comment by sln162** [comment]
@ShaunLoganOracle According to my observation, every time there is a major version update (such as 117), WebView2 can download to 117, and the bootstrapper can also install to 117, but the push update needs to wait for several days, unlike the minor version updates that are faster, sometimes it may be delayed for 10 days, maybe for security reasons, or maybe for batch push. I think, if you want to wait for your customers to solve this problem automatically, you need to wait for their machines to upgrade WebView2 automatically, which may take another two weeks.

**Comment by BenjaminGopp** [comment]
Deleting this entry worked for me!

![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/36400916/6a04ab2e-72fd-4bf4-bbe5-c64d4bf07f8b)

**Comment by nickdaria** [comment]
> Deleting this entry worked for me!
> 
> ![image](https://private-user-images.githubusercontent.com/36400916/300732024-6a04ab2e-72fd-4bf4-bbe5-c64d4bf07f8b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIzNTczNzMsIm5iZiI6MTcyMjM1NzA3MywicGF0aCI6Ii8zNjQwMDkxNi8zMDA3MzIwMjQtNmEwNGFiMmUtNzJmZC00YmY0LWJiZTUtYzY0ZDRiZjA3ZjhiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzMwVDE2MzExM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQyN2Q3ZWNmMDFiZjY2MTY4YWNhYjE4YzYwYjFiNmYyOWY2YzBlYmNkMDc2N2QyOGM5NWM5YjM3MDBlYzBkMDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hQ6osmJcjSU0wvH4cCmALWfdQBx5hpbCxGF35BqACfk)

It took me way longer than it needed to in order to find this. Saw hints to invalid config but was wondering where the hell it was coming from.

Thanks

**Comment by sln162** [comment]
> > 删除此条目对我有用！
> 
> 为了找到这个，我花了比需要的时间更长的时间。看到了无效配置的提示，但想知道它到底是从哪里来的。
> 
> 我迅速实施了此功能，以缓解问题，以防在现场发生。值得庆幸的是，密钥已添加到当前用户中，因此我们可以在没有管理员的情况下将其删除。这不会受到版本号更改的影响，
> 
> ```
> private void RepairFaultyRegistryCompatibilitySetting()
> {
>     const string problemFolder = @"Software\Microsoft\Windows NT\CurrentVersion\AppCompatFlags\Layers";
>     const string problemSuffix = "msedgewebview2.exe";
>     const string problemValue = "HIGHDPIAWARE";
> 
>     try
>     {
>         using (RegistryKey baseKey = Registry.CurrentUser.OpenSubKey(problemFolder, true))
>         {
>             if (baseKey == null)
>             {
>                 Debug.WriteLine($"Registry key not found: {problemFolder}");
>                 return;
>             }
> 
>             string[] valueNames = baseKey.GetValueNames();
> 
>             foreach (string valueName in valueNames)
>             {
>                 if (valueName.EndsWith(problemSuffix, StringComparison.OrdinalIgnoreCase))
>                 {
>                     object value = baseKey.GetValue(valueName);
>                     if (value != null && value.ToString().Contains(problemValue))
>                     {
>                         baseKey.DeleteValue(valueName);
>                         Debug.WriteLine($"Deleted registry key: {valueName}");
>                     }
>                 }
>             }
>         }
>     }
>     catch (Exception ex)
>     {
>         Debug.WriteLine($"An error occurred: {ex.Message}");
>     }
> }
> ```

I encountered a similar problem last year, and the solution is the same as yours. This inspiration came from LiangTheDev's reply above, where he mentioned the registry location multiple times. I don't know what tool added this registry record because my program monitored that it appeared again after being deleted for a period of time. So remember to check if this registry record was added every time initialization fails.

**Comment by ahmadsarkar** [comment]
To resolve the WebView2 initialization failure due to a DPI mismatch, I integrated the SetProcessDpiAwarenessContext(DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE_V2) function at the start of the wWinMain function. This ensures that the application's DPI awareness is consistent with the WebView2 processes, preventing errors when users change display scaling (e.g., to 125%). This adjustment resolved the DPI-related issue and allowed WebView2 to initialize correctly.

**Comment by Limanima** [comment]
Hello, I am this problem too. I've already confirmed that by removing the registry key the problem gets "resolved".
But, isn't this going to get fixed? Removing the key when the application starts is a nice workaround, but it is a workaround nonetheless. And that key does not belong to my application, it doesn't feel right go there and remove it. Thanks!

**Comment by ahmadsarkar** [comment]
> Hello, I have this problem too. I've already confirmed that by removing the registry key the problem gets "resolved". But, isn't this going to get fixed? Removing the key when the application starts is a nice workaround, but it is a workaround nonetheless. And that key does not belong to my application, it doesn't feel right go there and remove it. Thanks!

hi there . 
I have this problem in C++  (windows desktop application with C++)
So if you are using C++ just

Copy this line of code in your  wWinMain

  SetProcessDpiAwarenessContext(DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE_V2);


After windows 10 we have display scaling  and you can set this line to set display scaling 

If you use .net C# concept is same as .

**Comment by Limanima** [comment]
@ahmadsarkar I'm aware that I can do that, but the thing is:
Isn't that just another workaround to the problem? 
Beseides, our app is installed in about 3000 client machines, I'm not completely sure that setting DPI awareness is not going to change the "look" of the UI. We have some users that use 2 monitors, other just one, some use 125% display settings, others use 150%, actually we don't even know what settings the users have.
I'm not completely comfortable setting that parameter application wise.
I'm not also not clarified if this is in fact a bug on webview or not.

**Comment by ShaunLoganOracle** [comment]
@yunate 

I have another customer encountering this issue.  Much of the basic background from my initial post ([here](https://github.com/MicrosoftEdge/WebView2Feedback/issues/3008#issue-1467998533)) is still true. Latest info:
VSTO Add-in in Excel in C#
Winforms + WV2 SDK 1.0.2739.15
WV2 Runtime: 137.0.3296.68
Windows 10 22H2
.NET Framework 4.8

I checked the reg keys:
```
HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows NT\CurrentVersion\AppCompatFlags\Layers
HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Microsoft\Windows NT\CurrentVersion\AppCompatFlags\Layers
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\AppCompatFlags\Layers
```
Only shows OneDrive.exe

The Exception:
```
COMException: The group or resource is not in the correct state to perform the requested operation. (Exception from HRESULT: 0x8007139F)
Source: mscorlib
Stack: 
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHRInternal(Int32 errorCode, IntPtr errorInfo)
   at Microsoft.Web.WebView2.Core.CoreWebView2Environment.<CreateCoreWebView2ControllerAsync>d__102.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Web.WebView2.WinForms.WebView2.<InitCoreWebView2Async>d__25.MoveNext()

```
I cannot run the debugger on the customer computer; I did have them capture a WebView2 diagnostic log that shows some errors, for example these snippets:
```
. . .
[18804:23140:0613/120254.867:ERROR:base\win\win_util.cc:1600] failed to set WebView dpi awareness of 1: The operation completed successfully. (0x0)
. . .
[18804:23140:0613/120255.049:ERROR:chrome\browser\edge_embedded_browser\connection_impl.cc:194] Incompatible WebView configuration, current command line:
	"C:\Program Files (x86)\Microsoft\EdgeWebView\Application\137.0.3296.68\msedgewebview2.exe" --embedded-browser-webview=1 --webview-exe-name=EXCEL.EXE --webview-exe-version=16.0.10417.20012 --user-data-dir="C:\Users\<redacted>\AppData\Local\Oracle\ADF Desktop Integration\EBWebView" --noerrdialogs --embedded-browser-webview-dpi-awareness=1 --disable-features=msSmartScreenProtection --enable-features=msSingleSignOnOSForPrimaryAccountIsShared --enable-logging --log-file="c:\temp\webview2.log" --v=1 --lang=en-US --mojo-named-platform-channel-pipe=6116.13236.11007378707826275769 --flag-switches-begin --flag-switches-end --no-default-browser-check --disable-component-extensions-with-background-pages --no-first-run --disable-default-apps --internet-explorer-integration=none --js-flags=--expose-gc --disable-popup-blocking --disable-sync --sdsm-state=off --do-not-de-elevateincoming 
. . .
[23232:21508:0613/120255.141:ERROR:base\win\win_util.cc:1600] failed to set WebView dpi awareness of 1: The data area passed to a system call is too small. (0x7A)
. . .
[23232:21508:0613/120255.345:ERROR:ui\gl\direct_composition_support.cc:620] AMD VideoProcessorGetOutputExtension failed: Unspecified error (0x80004005)
[23232:21508:0613/120255.348:VERBOSE1:components\viz\service\main\viz_main_impl.cc:85] VizNullHypothesis is disabled (not a warning)
[23232:21508:0613/120255.353:WARNING:ui\gl\direct_composition_support.cc:1172] QueryInterface to IDCompositionDevice4 failed: No such interface supported (0x80004002)
. . .
```
I can provide the full log if needed.

Our add-in code that instantiates the WebView2 WinForms control does not set any DPI settings, so the command line to msedgewebview2.exe seen in the log above is being generated (with the `--embedded-browser-webview-dpi-awareness=1`) by  the WV2 SDK/loader.

Edit: Keeping in mind that our add-in is running within the Excel.exe process.

For other customers, the "fix" was to remove Excel.Exe from the `SOFTWARE\Microsoft\Windows NT\CurrentVersion\AppCompatFlags\Layers` registry keys.  That does  not apply here. Our joint (Oracle + Microsoft) customer is looking for answers - the inability to use WebView2 via the Excel add-in is blocking them.,

**Comment by ShaunLoganOracle** [comment]
@yunate @LiangTheDev 
I recognize that my latest update (10 days ago) was on a Closed thread and so might not be seen.  Shall I open a new Issue for my latest update? 

Summary: customer seeing the same exception as reported earlier, but no apparent DPI Awareness mismatch. However WV2 logging shows some ERROR entries with `Incompatible WebView configuration` and `[18804:23140:0613/120254.867:ERROR:base\win\win_util.cc:1600] failed to set WebView dpi awareness of 1: The operation completed successfully. (0x0)`
